recipes/newrelic/infrastructure/windows.yml

```diff
@@ -26,63 +26,26 @@ install:
         - task: assert_agent_started
 
-    remove_any_previous:
-      ignore_error: true
-      cmds:
-        - |
-          powershell -command '
-          function Find-UninstallGuids {
-            param (
-              [Parameter(Mandatory)]
-              [string]$Match
-            )
-
-            $baseKeys = Get-ChildItem -Path HKLM:\SOFTWARE\Microsoft\Windows\CurrentVersion\Uninstall `
-            | % { $_.Name.TrimStart("HKEY_LOCAL_MACHINE\") }
-
-            $wowKeys = Get-ChildItem -Path HKLM:\SOFTWARE\Wow6432Node\Microsoft\Windows\CurrentVersion\Uninstall `
-            | % { $_.Name.TrimStart("HKEY_LOCAL_MACHINE\") }
-
-            $allKeys = $baseKeys + $wowKeys
-
-            $uninstallIds = New-Object System.Collections.ArrayList
-            foreach ($key in $allKeys) {
-              $keyData = Get-Item -Path HKLM:\$key
-              $name = $keyData.GetValue("DisplayName")
-              if ($name -and $name -match $Match) {
-                $keyId = Split-Path $key -Leaf
-                $uninstallIds.Add($keyId) | Out-Null
-              }
-            }
-
-            if ($uninstallIds.Count -eq 0) {
-              return @()
-            }
-
-            return $uninstallIds
-          }
-
-          $uninstallIds = Find-UninstallGuids -Match "New Relic Infrastructure Agent"
-          foreach ($uninstallId in $uninstallIds) {
-            $uninstallCommand = "msiexec /x """ + $uninstallId + """ /qn"
-            Invoke-Expression $uninstallCommand
-            # wait here to allow uninstall command to complete since it returns without waiting
-            Start-Sleep -s 15
-          }
-          '
-
     install:
       cmds:
         - powershell -command '(New-Object System.Net.WebClient).DownloadFile("https://download.newrelic.com/infrastructure_agent/windows/newrelic-infra.msi", "$env:TEMP\newrelic-infra.msi");'
         - powershell -command '$LICENSE_KEY="{{.NEW_RELIC_LICENSE_KEY}}"; msiexec.exe /qn /i "$env:TEMP\newrelic-infra.msi" GENERATE_CONFIG=true LICENSE_KEY="$LICENSE_KEY" | Out-Null;'
         - powershell -command '(New-Object System.Net.WebClient).DownloadFile("https://aka.ms/vs/16/release/vc_redist.x64.exe", "$env:TEMP\vc_redist.x64.exe");'
         - powershell -command 'start-process -FilePath "$env:TEMP\vc_redist.x64.exe" -ArgumentList "/install /q /norestart" -Verb RunAs -wait | Out-Null;'
-        - powershell -command 'Add-Content -Path "C:\\Program Files\\New Relic\\newrelic-infra\\newrelic-infra.yml" -Value "enable_process_metrics`:` true" -Force;'
-        - powershell -command '$NEW_RELIC_REGION="{{.NEW_RELIC_REGION}}"; if ($NEW_RELIC_REGION -ilike "staging") { Add-Content -Path "C:\\Program Files\\New Relic\\newrelic-infra\\newrelic-infra.yml" -Value "staging`:` true" -Force; };'
+        - |
+          powershell -command '
+          $InfraConfig = "C:\\Program Files\\New Relic\\newrelic-infra\\newrelic-infra.yml"
+          if (-not (Test-Path $InfraConfig)) {
```

Comment:
@noahmmcgivern why the not? We should always create a new file if it's not there.
I believe we want to set the "enable_process_metrics" according to what the recipe says (true), we just don't want to override other values than the ones the install sets.
Similarly to linux, we'll also want to remove the "staging" element if it's not a staging deployment